instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/railtie.rb

```diff
@@ -7,24 +7,39 @@
 module OpenTelemetry
   module Instrumentation
     module ActionMailer
-      SUBSCRIPTIONS = %w[
-        deliver.action_mailer
-      ].freeze
+      DELIVER_SUBSCRIPTION = 'deliver.action_mailer'
+      PROCESS_SUBSCRIPTION = 'process.action_mailer'
 
       # This Railtie sets up subscriptions to relevant ActionMailer notifications
       class Railtie < ::Rails::Railtie
         config.after_initialize do
           ::OpenTelemetry::Instrumentation::ActiveSupport::Instrumentation.instance.install({})
+          subscribe_to_deliver
+          subscribe_to_process
+        end
 
-          SUBSCRIPTIONS.each do |subscription_name|
-            config = ActionMailer::Instrumentation.instance.config
+        class << self
+          def subscribe_to_deliver
```

Comment:
I moved these bits into class methods that the railtie callback calls. This way I can call them from the tests to subscribe (and unsubscribe) with different configurations.